numpy/core/shape_base.pyi

```diff
@@ -31,32 +31,61 @@ def atleast_3d(arys: ArrayLike, /) -> NDArray[Any]: ...
 def atleast_3d(*arys: ArrayLike) -> list[NDArray[Any]]: ...
 
 @overload
-def vstack(tup: Sequence[_ArrayLike[_SCT]]) -> NDArray[_SCT]: ...
+def vstack(
+    tup: Sequence[_ArrayLike[_SCT]],
+    *,
+    dtype: DTypeLike  = ..., 
+    casting: _CastingKind = ...
+) -> NDArray[_SCT]: ...
 @overload
-def vstack(tup: Sequence[ArrayLike]) -> NDArray[Any]: ...
+def vstack(
+    tup: Sequence[ArrayLike],
+    *,
+    dtype: DTypeLike  = ..., 
+    casting: _CastingKind = ...
+) -> NDArray[Any]: ...
 
 @overload
-def hstack(tup: Sequence[_ArrayLike[_SCT]]) -> NDArray[_SCT]: ...
+def hstack(
+    tup: Sequence[_ArrayLike[_SCT]], 
+    *,
+    dtype: DTypeLike  = ..., 
+    casting: _CastingKind = ...
+) -> NDArray[_SCT]: ...
 @overload
-def hstack(tup: Sequence[ArrayLike]) -> NDArray[Any]: ...
+def hstack(
+    tup: Sequence[ArrayLike], 
+    *,
+    dtype: DTypeLike  = ..., 
+    casting: _CastingKind = ...
+) -> NDArray[Any]: ...
 
 @overload
 def stack(
     arrays: Sequence[_ArrayLike[_SCT]],
     axis: SupportsIndex = ...,
-    out: None = ...,
+    out: DTypeLike  = ...,
```

Comment:
```suggestion
    out: None = ...,
```
@jhonatancunha I found the culprit. Seems the `out` type was accidentally changed.

Reply:
@BvB93 Sorry for that, I already pushed the correction for this mistake, but emerged a new error in the build process (No overload variant of "stack" matches argument types "List[List[int]]", "SubClass").

 I was looking at the overloads of concatenate method, and there are some cases that it is using DTypeLike, _DTypeLike and None on dtype keyword argument. Should I change the use of DTypeLike on every overload of stack method?